File: src/main/java/no/nav/veilarbaktivitet/aktivitetskort/bestilling/EksternAktivitetskortBestilling.java
package no.nav.veilarbaktivitet.aktivitetskort.bestilling;

import lombok.Getter;
import no.nav.veilarbaktivitet.aktivitet.domain.AktivitetData;
import no.nav.veilarbaktivitet.aktivitetskort.ActionType;
import no.nav.veilarbaktivitet.aktivitetskort.Aktivitetskort;
import no.nav.veilarbaktivitet.aktivitetskort.AktivitetskortMapper;
import no.nav.veilarbaktivitet.aktivitetskort.AktivitetskortType;
import no.nav.veilarbaktivitet.person.Person;

import java.util.UUID;

@Getter
public class EksternAktivitetskortBestilling extends AktivitetskortBestilling {

    public EksternAktivitetskortBestilling(Aktivitetskort aktivitetskort, String source, AktivitetskortType type, UUID messageId, ActionType actionType, Person.AktorId aktorId) {
        super(source, type, aktivitetskort, messageId, actionType, aktorId);
    }

    @Override
    public AktivitetData toAktivitet() {
        return AktivitetskortMapper.mapTilAktivitetData(this, null);
    }

    @Override
    public UUID getAktivitetskortId() {
        return this.getAktivitetskort().getId();
    }
}

File: src/main/kotlin/no/nav/veilarbaktivitet/aktivitetskort/bestilling/EksternAktivitetskortBestilling.kt
package no.nav.veilarbaktivitet.aktivitetskort.bestilling

import no.nav.veilarbaktivitet.aktivitetskort.ActionType
import no.nav.veilarbaktivitet.aktivitetskort.dto.Aktivitetskort
import no.nav.veilarbaktivitet.aktivitetskort.dto.AktivitetskortType
import no.nav.veilarbaktivitet.person.Person
import java.util.*

class EksternAktivitetskortBestilling(
    aktivitetskort: Aktivitetskort,
    source: String,
    type: AktivitetskortType,
    messageId: UUID,
    actionType: ActionType,
    aktorId: Person.AktorId
) : AktivitetskortBestilling(source, type, aktivitetskort, messageId, actionType, aktorId) {
    override fun getAktivitetskortId() = aktivitetskort.id

}